servicex/databinder/databinder_configuration.py

```diff
@@ -0,0 +1,257 @@
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+# DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+# FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+# DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+# SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+# OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+import yaml
+import pathlib
+from typing import Any, Dict, Union
+import rich
+
+
+def load_databinder_config(input_config:
+                           Union[str, pathlib.Path, Dict[str, Any]]
+                           ) -> Dict[str, Any]:
+    """
+    Loads, validates, and returns DataBinder configuration.
+    The order of function call matters.
+    Args:
+        input_config (Union[str, pathlib.Path, Dict[str, Any]]):
+            path to config file or config as dict
+    Returns:
+        Dict[str, Any]: configuration
+    """
+    def prepare_config(config):
+        _set_default_values(config)
+        ndef = _replace_definition_in_sample_block(config)
+        rich.print(f"Replaced {ndef} Option values from Definition block")
+        _support_old_option_names(config)
+        _decorate_backend_per_sample(config)
+        return _validate_config(config)
+
+    if isinstance(input_config, Dict):
+        return prepare_config(input_config)
+    else:
+        file_path = pathlib.Path(input_config)
+        config = yaml.safe_load(file_path.read_text())
+        rich.print(f"Loading DataBinder config file: {file_path}")
+        return prepare_config(config)
+
+
+def _set_default_values(config: Dict[str, Any]) -> Dict:
+    """
+    Default values in General block:
+        Delivery: LocalPath
+        OutputFormat: root
+    """
+    if 'Delivery' in config['General'].keys():
+        config['General']['Delivery'] = config['General']['Delivery'].lower()
+    else:
+        config['General']['Delivery'] = 'localpath'
+
+    if 'OutputFormat' not in config['General'].keys():
+        config['General']['OutputFormat'] = 'root'
+
+    if 'OutFilesetName' not in config['General'].keys():
+        config['General']['OutFilesetName'] = 'servicex_fileset'
+
+    for (idx, sample) in zip(range(len(config['Sample'])), config['Sample']):
+        if 'IgnoreLocalCache' not in sample.keys():
+            if 'IgnoreLocalCache' in config['General'].keys():
+                config['Sample'][idx]['IgnoreLocalCache'] = config['General']['IgnoreLocalCache']
+            else:
+                config['Sample'][idx]['IgnoreLocalCache'] = False
+
+    return config
+
+
+def _replace_definition_in_sample_block(config: Dict[str, Any]):
```

Comment:
Wow! What is this method doing? Can you write up a comment describing what is going on here?